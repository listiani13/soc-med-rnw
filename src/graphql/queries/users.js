// @flow
import gql from 'graphql-tag';

export const GET_USER_LIST = gql`
  query getUserList {
    data @rest(type: "Persons", path: "users") {
      id
      name
      username
      website
      phone
    }
  }
`;
