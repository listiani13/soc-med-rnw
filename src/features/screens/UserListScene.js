// @flow
import React, {Component} from 'react';
import {View} from 'react-native';
import {Text} from '../../core-ui';
import {USER_LIST} from '../../data/users';
import UserCard from '../../components/UserCard';

export default class UserList extends Component<{}> {
  render() {
    return (
      <View
        style={{
          flex: 1,
          paddingVertical: 20,
        }}
      >
        <Text size="xlarge" style={{paddingHorizontal: 10}}>
          User List
        </Text>
        <View
          style={{flexDirection: 'row', flexWrap: 'wrap', alignItems: 'center'}}
        >
          {USER_LIST.map((user, index) => {
            let {name, username, website, phone} = user;
            return (
              <UserCard
                name={name}
                username={username}
                website={website}
                phone={phone}
                style={{marginHorizontal: 12, marginVertical: 10}}
              />
            );
          })}
        </View>
      </View>
    );
  }
}
