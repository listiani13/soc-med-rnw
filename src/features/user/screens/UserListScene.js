// @flow
import React, {Component} from 'react';
import {View, StyleSheet} from 'react-native';
import {Route} from 'react-router-dom';
import {Text} from '../../../core-ui';
import {USER_LIST} from '../../../data/users';
import UserCard from '../../../components/UserCard';

export default class UserList extends Component<{}> {
  render() {
    return (
      <Route>
        {({history}) => {
          return (
            <View style={styles.root}>
              <Text size="xlarge" style={styles.title}>
                User List
              </Text>
              <View style={styles.content}>
                {USER_LIST.map((user, index) => {
                  let {name, username, website, phone, id} = user;
                  return (
                    <UserCard
                      name={name}
                      username={username}
                      website={website}
                      phone={phone}
                      style={styles.userCard}
                      key={index}
                      onClick={() => history.push(`/user/${id}`)}
                    />
                  );
                })}
              </View>
            </View>
          );
        }}
      </Route>
    );
  }
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    paddingVertical: 20,
  },
  title: {
    paddingHorizontal: 10,
  },
  content: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignItems: 'center',
  },
  userCard: {
    marginHorizontal: 12,
    marginVertical: 10,
  },
});
