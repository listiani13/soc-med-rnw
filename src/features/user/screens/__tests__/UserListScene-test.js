import React from 'react';
import {render} from 'react-testing-library';
import {HashRouter} from 'react-router-dom';
import {MockedProvider} from 'react-apollo/test-utils';
import UserListScene from '../UserListScene';
import {GET_USER_LIST} from '../../../../graphql/queries/users';
import wait from 'waait';

describe('UserListScene', () => {
  let mocks = [
    {
      request: {
        query: GET_USER_LIST,
      },
      result: {
        data: {
          data: [
            {
              id: 1,
              name: 'Leanne Graham',
              username: 'Bret',
              email: 'Sincere@april.biz',
              phone: '1-770-736-8031 x56442',
              website: 'hildegard.org',
            },
          ],
        },
      },
    },
  ];

  it('renders without crashing', async () => {
    let {container} = render(
      <HashRouter>
        <MockedProvider mocks={mocks} addTypename={false}>
          <UserListScene />
        </MockedProvider>
      </HashRouter>,
    );
    // NOTE: Wait to get first query result
    await wait(10);
    expect(container).toMatchSnapshot();
  });
});
