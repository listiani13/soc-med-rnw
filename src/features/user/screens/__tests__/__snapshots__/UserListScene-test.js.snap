// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListScene renders without crashing 1`] = `
<div>
  <div
    class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexGrow-16y2uox rn-flexShrink-1wbh5a2 rn-flexBasis-1ro0kt6 rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-1knelpx rn-paddingRight-9aemit rn-paddingBottom-k8qxaj rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-1qfoi16 rn-paddingBottom-1mdbw0j rn-paddingLeft-1hfyk0a rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
      style="font-size: 36px;"
    >
      User List
    </div>
    <div
      class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-18u37iz rn-flexWrap-1w6e6rj rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
    >
      <div
        class="rn-alignItems-1oszu61 rn-borderTopColor-478rc6 rn-borderRightColor-bmvd0i rn-borderBottomColor-y0r23w rn-borderLeftColor-12jqpwx rn-borderTopLeftRadius-kicko2 rn-borderTopRightRadius-notknq rn-borderBottomRightRadius-1dpl46z rn-borderBottomLeftRadius-6t5ypu rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-youum4 rn-borderRightWidth-15ik8dz rn-borderBottomWidth-9cy4zw rn-borderLeftWidth-1mdj7ya rn-boxShadow-18cwage rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-height-1kl1mp1 rn-marginTop-156q2ks rn-marginRight-1b7u577 rn-marginBottom-15d164r rn-marginLeft-19u6a5r rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-touchAction-1otgn73 rn-transitionDuration-eafdt9 rn-transitionProperty-1i6wzkk rn-userSelect-lrvibr rn-width-l0gwng rn-zIndex-1lgpqti"
        data-focusable="true"
        tabindex="0"
      >
        <div
          class="rn-alignItems-1awozwy rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-1knelpx rn-paddingRight-j2kj52 rn-paddingBottom-k8qxaj rn-paddingLeft-f727ji rn-position-bnwqim rn-zIndex-1lgpqti"
        >
          <div
            class="MuiAvatar-root-1 MuiAvatar-colorDefault-2"
            style="width: 90px; height: 90px; margin-bottom: 10px;"
          >
            LG
          </div>
          <div
            class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1i10wst rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-d0pm55 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textAlign-q4m81j rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
            style="font-weight: 500;"
          >
            Leanne Graham
          </div>
          <div
            class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-d0pm55 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
            style="font-size: 13px;"
          >
            Bret
          </div>
          <div
            class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-d0pm55 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
            style="font-size: 13px;"
          >
            1-770-736-8031 x56442
          </div>
          <div
            class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginLeft-11wrixw rn-marginBottom-d0pm55 rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
            dir="auto"
            style="font-size: 13px;"
          >
            hildegard.org
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
